Add a getting started guide (#35)
nirmata/kyverno-aws-adapter: docs/getting_started.md

# Getting Started

This is a guide on how to get started with the Nirmata Kyverno Adapter for AWS. To learn more about Kyverno, check out the official documentation.

## Prerequisites
- a running EKS Cluster (refer [Creating an Amazon EKS Cluster](https://docs.aws.amazon.com/eks/latest/userguide/create-cluster.html))
- AWS CLI (refer [Installing the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html))
- eksctl (refer [Installing eksctl](https://docs.aws.amazon.com/eks/latest/userguide/eksctl.html))
- Helm CLI (refer [Installing Helm](https://helm.sh/docs/intro/install/))

## Installing the Nirmata Kyverno Adapter for AWS
There are a few steps we need to follow before installing the Helm chart.

### Creating the IAM Policy
To fetch the EKS Cluster configuration, the AWS Adapter needs the below permissions that has to be expressed via an IAM Policy

> ``bash

Reviewer: Missing backtick which breaks the code formatting
```suggestion
```bash
```

cat >my-policy.json <<EOF
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "eks:AccessKubernetesApi",
                  "eks:DescribeAddon",
                  "eks:DescribeAddonVersions",
                  "eks:DescribeCluster",
                  "eks:DescribeFargateProfile",
                  "eks:DescribeIdentityProviderConfig",
                  "eks:DescribeNodegroup",
                  "eks:DescribeUpdate",
                  "eks:ListAddons",
                  "eks:ListClusters",
                  "eks:ListFargateProfiles",
                  "eks:ListIdentityProviderConfigs",
                  "eks:ListNodegroups",
                "eks:ListTagsForResource",
                "eks:ListUpdates"
            ],
            "Resource": [
                "arn:aws:eks:*:844333597536:identityproviderconfig/*/*/*/*",
                "arn:aws:eks:*:844333597536:fargateprofile/*/*/*",
                "arn:aws:eks:*:844333597536:nodegroup/*/*/*",
                "arn:aws:eks:*:844333597536:cluster/*",

>                 "arn:aws:eks:*:844333597536:addon/*/*/*"

Author: Good catch!

```

### Creating the IAM Role
Create an IAM Role that references the policy we created earlier.

**Note:** We will use `eksctl` in this guide to create the IAM Role. The serviceaccount creation will be done by the Helm chart so that appropriate RBAC is assigned by the chart itself. If you wish to use your own serviceaccount, then make sure all the necessary rolebinding and clusterrolebinding are present.

```bash
account_id=$(aws sts get-caller-identity --query "Account" --output text)

eksctl create iamserviceaccount --name nirmata-aws-adapter  --namespace nirmata-aws-adapter --cluster <cluster-name> --role-name nirmata-adapter-role --attach-policy-arn arn:aws:iam::$account_id:policy/kyverno-aws-adapter-policy   --role-only  --approve
```

This will create a new IAM Role that references the policy we created above. This also creates the trust-relationship for us. If you wish to create the IAM Role via the management console or the AWS CLI, make sure to create the trust-relationship so that the AWS Adapter can assume this Role to fetch EKS Cluster details.

### Installing the AWS Adapter Helm chart

> First we need to set the variables that are needed to install the chart. You can either pass the arguments via the command line or set them in a file.

Reviewer: This might be helpful for Helm beginners 👀
```suggestion
First we need to set the values that are needed to install the Helm chart. You can either pass them as arguments via the [command line](https://helm.sh/docs/helm/helm_install/#helm-install) or set them in a [values file](https://helm.sh/docs/chart_template_guide/values_files/).
```

Here is the minimal info that you need to set
```bash
> cat myvalues.yaml
eksCluster:
  name: cluster-name
  region: cluster-region

> roleArn: arn:aws:iam::account-id:role/nirmata-adapter-role

Reviewer: A placeholder value seems more appropriate since we're showing a sample values file here
```suggestion
roleArn: arn:aws:iam::111122223333:role/nirmata-adapter-role
```
